### PR TITLE
[Feature] Support flip with label mapping

### LINKED ISSUE #590: Support `Flip` with `label transform`
In some datasets, the label is directional, such as `push left`. In this case, the `flip` augmentation cannot be applied.

But we can add `label transform` to `flip`, change the label of the flipped image with the label `push left` to `push right`, so as to augment the data.

### PR DESCRIPTION
**Pull request by congee524**
fix #590 

I'm doing a simple benchmark with TSM-R50 on sthv1

### CODE REVIEW

File: mmaction/datasets/pipelines/augmentations.py [hunk 1]
```diff
@@ -1120,6 +1127,11 @@ def __call__(self, results):
         results['flip'] = flip
         results['flip_direction'] = self.direction
 
+        if self.label_transforms is not None and flip:
+            label = results['label']
+            if label in self.label_transforms.keys():
```
Comment (dreamerlin) [inline mmaction/datasets/pipelines/augmentations.py:1132]: `if label in self.label_transforms:` is OK

File: mmaction/datasets/pipelines/augmentations.py [hunk 2]
```diff
@@ -1120,6 +1127,11 @@ def __call__(self, results):
         results['flip'] = flip
         results['flip_direction'] = self.direction
 
+        if self.label_transforms is not None and flip:
+            label = results['label']
```
Comment (innerlee) [inline mmaction/datasets/pipelines/augmentations.py:1131]:
```python
results['label'] = self.label_transforms.get(label, results['label'])
```

Comment (innerlee) [reply mmaction/datasets/pipelines/augmentations.py:1131]: okay my fault, i actually meant `results['label'] = self.label_transforms.get(results['label'], results['label']))` to save one line. sigh...

Comment (congee524) [reply mmaction/datasets/pipelines/augmentations.py]: I thought it would make the code easier to read.....

**Review by innerlee — APPROVED** [review]

**Comment by innerlee** [comment]
The code looks ok. @congee524 do you want to post the ckpt?

**Comment by congee524** [comment]
> The code looks ok. @congee524 do you want to post the ckpt?

yeah, and I'm working on it.

File: mmaction/datasets/pipelines/augmentations.py [hunk 3]
```diff
@@ -1091,16 +1091,23 @@ class Flip:
         flip_ratio (float): Probability of implementing flip. Default: 0.5.
         direction (str): Flip imgs horizontally or vertically. Options are
             "horizontal" | "vertical". Default: "horizontal".
+        label_transforms (Dict[int, int] | None): Transform the label of the
```
Comment (innerlee) [inline mmaction/datasets/pipelines/augmentations.py:1094]: Does `label_transforms` have better naming alternatives? `flipped_labels`? `flip_label_map`?

Comment (congee524) [reply mmaction/datasets/pipelines/augmentations.py:1094]: I prefer `flip_label_map`.

**Comment by dreamerlin** [comment]
`def __repr__(self):` should also be modified

**Review by dreamerlin — APPROVED** [review]

**Review by innerlee — APPROVED** [review]

**Pull request merged**